src/pyasm/prod/service/api_xmlrpc.py

```diff
@@ -924,6 +924,46 @@ def log(my, ticket, level, message, category="default"):
         return True
 
 
+
+
+    #
+    # Preferences
+    #
+    @xmlrpc_decorator
+    def get_preference(my, ticket, key):
+        '''Get the users preference for this project
+
+        @params
+        ticket - authentication ticket
+        key - unique key to identify preference
+
+        @return
+        current value of preference
+        '''
+        project_code = Project.get_project_code()
```

Comment:
project_code should be passed in and default to ''.. since all the existing preferences set in pref_setting don't have project_code set.